### PR TITLE
Add spotify-auth sample

### PR DESCRIPTION
**Pull request by andresgottlieb**

### CODE REVIEW

**Comment by prescottprue** [comment]
Nice example

**Comment by nicolasgarnier** [comment]
Thanks for the sample @andresgottlieb,  Sorry for not reviewing it right now as I'm cranking on some stuff for Google IO. I'll review as soon as I can find some time :)

File: spotify-auth/README.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
+# Use Spotify Sign In with Firebase
+
+This sample shows how to authenticate using Spotify Sign-In on Firebase. In this sample we use OAuth 2.0 based authentication to get Spotify user information then create a Firebase Custom Token (using the Spotify user ID).
+
+
+## Setup the sample
+
+Create and setup the Firebase project:
+ 1. Create a Firebase project using the [Firebase Developer Console](https://console.firebase.google.com).
+ 1. Enable Billing on your Firebase the project by switching to the **Blaze** plan, this is currently needed to be able to perform HTTP requests to external services from a Cloud Function.
```
Comment (nicolasgarnier) [inline spotify-auth/README.md]: typo: remove "the"

Comment (nicolasgarnier) [inline spotify-auth/README.md]: Here the developer can switch to the Blaze or the Flame plan. Can you also link to the pricing page: https://firebase.google.com/pricing/ please?

File: spotify-auth/functions/index.js [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+
+const OAUTH_SCOPES = ['user-read-email'];
+
+/**
+ * Creates a configured Spotify API Client instance.
+ */
+function spotifyClient() {
+  // Spotify OAuth 2 setup
+  // TODO: Configure the `spotify.client_id` and `spotify.client_secret` Google Cloud environment variables.
+  var SpotifyWebApi = require('spotify-web-api-node');
```
Comment (nicolasgarnier) [inline spotify-auth/functions/index.js]: Change `var` to `const`. Also you could move all this to the static init section :)

**Comment by nicolasgarnier** [comment]
ping :)

**Comment by nicolasgarnier** [comment]
OK I applied the styling fixes. Merging!

**Pull request merged**